pkg/scheduler/api/test_utils.go

```diff
@@ -94,10 +94,16 @@ type ScalarResource struct {
 
 // BuildResourceList builds resource list object
 func BuildResourceList(cpu string, memory string, scalarResources ...ScalarResource) v1.ResourceList {
-	resourceList := v1.ResourceList{
-		v1.ResourceCPU:    resource.MustParse(cpu),
-		v1.ResourceMemory: resource.MustParse(memory),
+	resourceList := v1.ResourceList{}
+
+	if len(cpu) > 0 {
+		resourceList[v1.ResourceCPU] = resource.MustParse(cpu)
 	}
+
+	if len(memory) > 0 {
+		resourceList[v1.ResourceMemory] = resource.MustParse(memory)
+	}
+
```

Comment:
does scalarResources has same issue?

Reply:
The modification here is just for the convenience of using in UT without passing cpu or memory. You can see that scalarResources is a slice. If it is an empty slice or nil, resourceList will not be assigned.